packages/api/pages/api/languages/[code]/import.ts

```diff
@@ -14,15 +19,103 @@ export default createRoute()
       import: z.string(),
     }),
     async handler(req, res) {
-      // TODO: implement import logic.
-      console.log('import language:', req.body.import);
-      // simulate the processing time
-      await new Promise<void>((resolve) => {
-        setTimeout(() => {
-          resolve();
-        }, 10000);
+      const language = await client.language.findUnique({
+        where: {
+          code: req.query.code,
+        },
       });
+
+      if (!language) {
+        res.notFound();
+        return;
+      }
+
+      // Delete all the glosses for the language.
+      await client.gloss.deleteMany({
+        where: {
+          languageId: language.id,
+        },
+      });
+
+      const glossData = [];
+
+      const start = new Date();
+      console.log('START:', start);
+      // TODO: we only use a few book keys for testing, to avoid creating a
+      //       ton of requests. use all keys when things are working
+      for (const key of bookKeys) {
+        console.log(key);
+        const importUrl = `${importServer}/${req.body.import}Glosses/${key}Gloss.js`;
+        const response = await fetch(importUrl);
+        const jsCode = await response.text();
+        const bookData = parseGlossJs(jsCode);
+        const bookId = bookKeys.indexOf(key) + 1;
+        const referenceData =
+          require(`../../../../../../data/morphology/${key}`) as string[][][];
```

Comment:
I had to temporarily disable the pre-commit hook to ignore the linter error about using `require`. How did you make it pass in `data/morphology.ts`?

Reply:
A dynamic import like that isn't going to work with a bundler. Instead we should update `data/morphology.ts` to import all of the json files and export as an object that maps the book key to the data. That way the bundler can include all of the data files in the build

Reply:
Ok!

Reply:
Done